part-i-getting-started/quick-start.md

```diff
@@ -0,0 +1,355 @@
+
+As our domain we'll take a Giftcard management. See the [wikipedia article](https://en.wikipedia.org/wiki/Gift_card) for a basic definition of gift cards. Essentially, there are just two events in the life cycle of a gift card:                                               
+* They get issued: a new gift card gets created with some amount of money stored.
+* They get redeemed: all or part of the monetary value stored on the gift card is used to purchase something.
+
+### Commands and Events
+As mentioned, we need a command to issue the Giftcard. This command will contain the identifier of the Giftcard and the initial amount. Let's build one:
+
+```java
+public class IssueCmd {
+
+    private final String id;
+    private final Integer amount;
+    
+    public IssueCmd(String id, Integer amount) {
+        this.id = id;
+        this.amount = amount;
+    }
+    
+    public String getId() {
+        return id;
+    }
+    
+    public Integer getAmount() {
+        return amount;
+    }
+}
+```
+
+We'll name an event that will be triggered after this command is successfuly processed `IssuedEvt`:
+
+```java
+public class IssuedEvt {
+
+    private final String id;
+    private final Integer amount;
+    
+    public IssuedEvt(String id, Integer amount) {
+        this.id = id;
+        this.amount = amount;
+    }
+    
+    public String getId() {
+        return id;
+    }
+    
+    public Integer getAmount() {
+        return amount;
+    }
+}
+```
+
+Similarly, we'll create `RedeemCmd` (do note that `amount` is amount to be deducted from the Giftcard) and `RedeemEvt`:
+
+```java
+public class RedeemCmd {
+
+    @TargetAggregateIdentifier // (1)
+    private final String id;
+    private final Integer amount;
+    
+    public RedeemCmd(String id, Integer amount) {
+        this.id = id;
+        this.amount = amount;
+    }
+    
+    public String getId() {
+        return id;
+    }
+        
+    public Integer getAmount() {
+        return amount;
+    }
+}
+```
+
+(1) `@TargetAggregateIdentifier` annotation is used by Axon to find the correct Giftcard aggregate instance.
+
+```java
+public class RedeemedEvt {
+    
+    private final String id;
+    private final Integer amount;
+    
+    public RedeemedEvt(String id, Integer amount) {
+        this.id = id;
+        this.amount = amount;
+    }
+    
+    public String getId() {
+        return id;
+    }
+        
+    public Integer getAmount() {
+        return amount;
+    }
+}
+```
+
+> **Note** Command/Event definition looks a bit cumbersome, this can be avoided using hierarchy between commands/events, or using libraries like [lombok](https://projectlombok.org/), or using [kotlin](https://kotlinlang.org/) which provides more concise way of defining commands/events.
+
+### Aggregate
+
+Now that we have our commands and events which represent the API of our application, we can create Giftcard aggregate:
+
+```java
+public class GiftCard {
```

Comment:
I'd personally recommend adding the `@AggregateRoot` as a class level annotation here. It is not mandatory though..what would your personal preference be @m1l4n54v1c ?

Reply:
To be honest, I don't see a value unless we want to specify a different aggregate type. It is unnecessary.

Reply:
Daaawh. I know you have a point, but my gut feel isn't happy about it.
Nonetheless, let's leave it like this.